### PR TITLE
UPBGE: Fix crash when we call Texture_Init whereas the object has no texture

### PR DESCRIPTION
**Pull request by youle31**
test file: http://pasteall.org/blend/index.php?id=43536

I'm not sure about the error message...

### CODE REVIEW

**Comment by panzergame** [comment]
The crash happen only when texID is negative or bigger than 7 ?

File: source/gameengine/VideoTexture/Texture.cpp [hunk 1]
```diff
@@ -242,6 +242,10 @@ static int Texture_init(Texture *self, PyObject *args, PyObject *kwds)
 			{
 				// get blender material texture
 				self->m_matTexture = mat->GetTexture(texID);
+				if (!self->m_matTexture) {
```
Comment (panzergame) [inline source/gameengine/VideoTexture/Texture.cpp]: Apparently a

**Comment by youle31** [comment]
A debug message could be usefull if we forgot to put a texture in the slot, but it is as you want.

**Comment by panzergame** [comment]
Yeah, maybe we can reuse the try catch but with an other message error. You can look at the THRWEXCP(MaterialNotAvail, S_OK);.
You can define an error named like TextureNotAvail.

**Comment by panzergame** [comment]
ge_keyboard_sensor http://download.upbge.fr/download/buildbot/branch/

**Comment by youle31** [comment]
This: http://pasteall.org/79288/diff doesn't seem to work. There is no crash but no message in the console.

**Comment by panzergame** [comment]
I think you have to some lines in Exception.h/cpp.
e.g:
```
extern ExpDesc TextureNotAvailDesc;
```
```
registerAllExceptions()
…
	TextureNotAvailDesc.registerDesc();
…

**Comment by youle31** [comment]
Ah yes sorry I forgot it. So I tried but I still get the same error message:
Python script error - object 'Plane', controller 'Python':
Traceback (most recent call last):
  File "Text", line 10, in <module>
RuntimeError

No decription is printed... Weird. Or I missed something somewhere.

**Comment by youle31** [comment]
Ok I understand it is not printed in debug mode but it is printed in release mode... So you can merge if you want

**Pull request closed without merging**